File: 백준/Silver/2004. 조합 0의 개수/조합 0의 개수.java
import java.io.*;
import java.util.*;

public class Main {
    static int factorial(int n, int p){
        int cnt = 0;
        while(n>=p){
            cnt += n/p;
            n/=p;
        }
        return cnt;
    }
    static int zero(int n, int m){
        int two = factorial(n,2)-factorial(m,2)-factorial(n-m,2);
        int five = factorial(n,5)-factorial(m,5)-factorial(n-m,5);
        return Math.min(two,five);
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        System.out.println(zero(n,m));
    }    
    
    
}
    

File: 백준/Silver/2004. 조합 0의 개수/조합 0의 개수.py
import sys
input = sys.stdin.readline


def factorial(n,p):
    cnt = 0
    while n>=p:
        cnt += n//p
        n//=p
    return cnt
def zero(n,m):
    cnt_2 = factorial(n,2) - factorial(m,2)- factorial(n-m,2)
    cnt_5 = factorial(n,5) - factorial(m,5)- factorial(n-m,5)
    return min(cnt_2,cnt_5)

n,m  = map(int,input().split())

print(zero(n,m))